api/login/login.go

```diff
@@ -2,19 +2,60 @@ package login
 
 import (
 	"api/database"
+	"os"
+	"time"
 
 	"encoding/json"
 	"fmt"
-	"github.com/gorilla/mux"
 	"log"
 	"net/http"
+
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/gorilla/mux"
 )
 
 type Account struct {
 	Email    string `json:"email"`
 	Password string `json:"password"`
 }
 
+// Get the user a JWT token is for, or an error if the token is invalid
+func ValidateToken(tokenString string) (string, error) {
+	token, err := jwt.ParseWithClaims(tokenString, &jwt.RegisteredClaims{}, func(token *jwt.Token) (interface{}, error) {
+		return []byte(os.Getenv("JWT_SECRET")), nil
+	})
+
+	if err != nil {
+		return "", err
+	}
+
+	claims, ok := token.Claims.(*jwt.RegisteredClaims)
+	if !ok || !token.Valid || time.Now().After(claims.ExpiresAt.Time) {
+		return "", fmt.Errorf("invalid token")
+	}
+
+	return claims.Subject, nil
+}
+
+// Get a JWT token for a user, or an error if the user is invalid
+func GetToken(username string, password string) (*jwt.Token, error) {
```

Comment:
Rename this to email for consistency and clarity, we haven't introduced username yet

Reply:
Username will be separate from email, and will be what is publicly shown on your profile